### PR TITLE
Extend kmstools to get Encryption Context from CLI

### PR DESCRIPTION
**Pull request by atanzu**
This commit modifies `kmstool-instance` to be able to pass Encryption context from command line arguments. Context shall be passed using the new `--encryption-context KEY=VAL` argument. User can specify multiple key-value pairs via an invocation like `--encryption-context KEY1=VALUE1 --encryption-context KEY2=VALUE2`.

Application `kmstool-enclave` has been modified to process Encryption context from the received command, and pass it further to the KMS API.

Issue #143

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,6 +91,7 @@ if (NOT WIN32)
     set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION 0unstable)
 
     target_compile_definitions(${PROJECT_NAME} PUBLIC)
+    target_compile_options(${PROJECT_NAME} PRIVATE "-Wall" "-Werror" "-Wpedantic")
```
Comment (eugkoira) [inline CMakeLists.txt:94]: would be better to have it in a separate commit

**Review by eugkoira — DISMISSED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (atanzu) [reply CMakeLists.txt:94]: Only for the SDK library, or for the library and the binaries?

**Comment by jplock** [comment]
Can/will the same change be done for `kmstool-enclave-cli`?

**Comment by atanzu** [comment]
> Can/will the same change be done for `kmstool-enclave-cli`?

Can do, but probably in a separate PR. The reason is, Encryption contexts should also supported for the `genkey` command, which means, that `aws_kms_generate_data_key_blocking` API needs to be also extended to support encryption contexts.

File: CMakeLists.txt (hunk 1 above)
Comment (atanzu) [reply CMakeLists.txt:94]: Re-pushed the branch, making this line a separate commit.

Comment (eugkoira) [reply CMakeLists.txt:94]: meant for both, but was more a nit

**Review by eugkoira — APPROVED** [review]

**Review by foersleo — APPROVED** [review]
